File: src/main/java/ua/softgroup/matrix/server/persistent/entity/WindowTime.java
package ua.softgroup.matrix.server.persistent.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import java.time.LocalTime;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
@Entity
public class WindowTime extends AbstractEntity<Long> {
    private static final long serialVersionUID = -3216454660635129210L;

    @Column(columnDefinition = "TEXT")
    private String windowTitle;

    @Column
    private LocalTime startTime;

    @Column
    private Integer time = 0;

    @ManyToOne
    private TrackingData trackingData;

    public WindowTime() {
    }

    public WindowTime(String windowTitle, LocalTime startTime, Integer time, TrackingData trackingData) {
        this.windowTitle = windowTitle;
        this.startTime = startTime;
        this.time = time;
        this.trackingData = trackingData;
    }

    public String getWindowTitle() {
        return windowTitle;
    }

    public void setWindowTitle(String windowTitle) {
        this.windowTitle = windowTitle;
    }

    public LocalTime getStartTime() {
        return startTime;
    }

    public void setStartTime(LocalTime startTime) {
        this.startTime = startTime;
    }

    public Integer getTime() {
        return time;
    }

    public void setTime(Integer time) {
        this.time = time;
    }

    public TrackingData getTrackingData() {
        return trackingData;
    }

    public void setTrackingData(TrackingData trackingData) {
        this.trackingData = trackingData;
    }
}

File: src/main/scala/ua/softgroup/matrix/server/persistent/entity/WindowTime.scala
package ua.softgroup.matrix.server.persistent.entity

import java.time.LocalTime
import javax.persistence.{Column, Entity, ManyToOne}

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
@Entity
@SerialVersionUID(-3216454660635129210L)
class WindowTime extends AbstractEntity[java.lang.Long] {

  @Column(columnDefinition = "TEXT")
  @BeanProperty
  var windowTitle: String = _

  @Column
  @BeanProperty
  var startTime: LocalTime = _

  @Column
  @BeanProperty
  var time: Int = _

  @ManyToOne
  @BeanProperty
  var trackingData: TrackingData = _

  def this(windowTitle: String, startTime: LocalTime, time: Int, trackingData: TrackingData) {
    this()
    this.windowTitle = windowTitle
    this.startTime = startTime
    this.time= time
    this.trackingData= trackingData
  }

  override def toString = s"WindowTime(id=${super.getId}, windowTitle=$windowTitle, startTime=$startTime, time=$time)"
}
